src/main.rs

```diff
@@ -1649,6 +1657,76 @@ fn cmd_regenerate_audit_as(
     Ok(())
 }
 
+fn cmd_renew(out: &Arc<dyn Out>, cfg: &Config, sub_args: &RenewArgs) -> Result<(), miette::Report> {
+    trace!("renewing wildcard audits");
+    let mut store = Store::acquire_offline(cfg)?;
+    do_cmd_renew(out, &mut store, sub_args);
+    store.commit()?;
+    Ok(())
+}
+
+fn do_cmd_renew(out: &Arc<dyn Out>, store: &mut Store, sub_args: &RenewArgs) {
+    assert!(sub_args.expiring ^ sub_args.crate_name.is_some());
+
+    let today = <chrono::DateTime<chrono::Utc>>::from(SystemTime::now()).date_naive();
+    let new_end_date = today + chrono::Months::new(12);
+
+    let renew_entry = |entry: &mut WildcardEntry| {
+        entry.end = new_end_date.into();
+    };
+
+    if let Some(name) = &sub_args.crate_name {
+        match store.audits.wildcard_audits.get_mut(name) {
+            Some(audits) => {
+                audits.iter_mut().for_each(renew_entry);
+                writeln!(
+                    out,
+                    "Updated wildcard audits for {name} to expire on {new_end_date}"
+                );
+            }
+            None => {
+                warn!("ran `renew {name}`, but there are no wildcard audits for the crate");
+            }
+        }
+    } else {
+        // Find and update all expiring crates.
+        assert!(sub_args.expiring);
+        let expire_date = today + *WILDCARD_AUDIT_EXPIRATION_DURATION;
+
+        let mut updated = Vec::new();
+        for (name, entries) in store.audits.wildcard_audits.iter_mut() {
+            let any = entries
+                .iter_mut()
+                .filter(|e| e.should_suggest_renewal(expire_date))
+                .any(|e| {
+                    renew_entry(e);
+                    true
+                });
+            if any {
+                updated.push(name);
+            }
+        }
+
+        if updated.is_empty() {
+            info!("no wildcard audits are expiring in the next {WILDCARD_AUDIT_EXPIRATION_STRING}");
+            return;
+        }
+
+        updated.sort();
+
+        write!(
+            out,
+            "Updated wildcard audits for the following crates to expire on {new_end_date}:"
+        );
+        let mut iter = updated.iter();
+        write!(out, " {}", iter.next().unwrap());
+        for name in iter {
+            write!(out, ", {name}");
+        }
+        writeln!(out);
```

Comment:
Was this intended to use the new list formatting code you added?

Perhaps it'd make sense to do one crate on each line, and also write out who the publisher(s) are which you extended the wildcard audit(s) for?

Reply:
Ah good catch, I had intended to change that to be one per line. Including the publisher(s) would be a good addition.